prometheus/exporter/intervalpush/exporter.go

```diff
@@ -39,10 +42,20 @@ func (e *exporter) Export(exitChan <-chan bool) {
 		select {
 		case <-exitChan:
 			return
+		case w := <-e.done:
+			e.t.Stop()
+			w <- e.p.Push()
+			return
 		case <-e.t.C:
 			if err := e.p.Push(); err != nil {
 				log.Noticef("Push to gatherer failed: %s", err.Error())
 			}
 		}
 	}
 }
+
+func (e *exporter) Close() error {
+	w := make(chan error)
+	e.done <- w
+	return <-w
```

Comment:
would it be simpler like this
```go ?
e.done <- struct{}
e.t.Stop()
return e.Push()
```

Reply:
By doing this you don't have any guarantee that the pusher goroutine is actually stopped before pushing. Might not have any side effects as both the pusher and the ticker are reentrant, but I would rather stick to my version